src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -1159,6 +1167,12 @@ private void UpdateOrCreateLevel(long number, BlockInfo blockInfo)
                     level = new ChainLevelInfo(false, new[] {blockInfo});
                 }
 
+                if (setAsMain)
+                {
+                    level.HasBlockOnMainChain = true;
+                    blockInfo.WasProcessed = true;
```

Comment:
risky... if ! should process then mark as WasProcessed - can we just set HasBlockOnMainChain without WasProcessed? (it may cause a lot of trouble in some post fast sync deep reorgs, I guess